patches/tock/10-connect-vendor-hid-usb-interface.patch

```diff
@@ -0,0 +1,304 @@
+                     // receive data on the Interrupt OUT endpoint.
+-                    self.controller().endpoint_resume_out(1);
++                    self.controller().endpoint_resume_out(endpoint);
+                 }
+             }
+             true
+@@ -280,14 +298,17 @@ impl<'a, 'b, C: hil::usb::UsbController<'a>> ClientCtapHID<'a, 'b, C> {
+     // Send an OUT packet available in the controller back to the client.
+     // This returns false if the client is not ready to receive a packet, and true if the client
+     // successfully accepted the packet.
+-    fn send_packet_to_client(&'a self) -> bool {
++    fn send_packet_to_client(&'a self, endpoint: usize) -> bool {
+         // Copy the packet into a buffer to send to the client.
+         let mut buf: [u8; 64] = [0; 64];
+-        for (i, x) in self.out_buffer.buf.iter().enumerate() {
++        if endpoint == 0 || endpoint > NUM_ENDPOINTS {
++            return false
++        }
++        for (i, x) in self.out_buffer[endpoint-1].buf.iter().enumerate() {
+             buf[i] = x.get();
+         }
+ 
+-        assert!(!self.delayed_out.get());
++        assert!(self.delayed_out.is_none());
+ 
+         // Notify the client
+         if self
+@@ -301,11 +322,11 @@ impl<'a, 'b, C: hil::usb::UsbController<'a>> ClientCtapHID<'a, 'b, C> {
+             // should be re-transmitted or not.
+             self.cancel_in_transaction();
+ 
+-            self.client.map(|client| client.packet_received(&buf));
++            self.client.map(|client| client.packet_received(&buf, endpoint));
+             true
+         } else {
+             // Cannot receive now, indicate a delay to the controller.
+-            self.delayed_out.set(true);
++            self.delayed_out.set(endpoint);
+             false
+         }
+     }
+@@ -336,11 +357,21 @@ impl<'a, 'b, C: hil::usb::UsbController<'a>> hil::usb::Client<'a> for ClientCtap
+ 
+         // Set up the interrupt in-out endpoint
+         self.controller()
+-            .endpoint_set_in_buffer(1, &self.in_buffer.buf);
++            .endpoint_set_in_buffer(ENDPOINT_NUM, &self.in_buffer[0].buf);
+         self.controller()
+-            .endpoint_set_out_buffer(1, &self.out_buffer.buf);
++            .endpoint_set_out_buffer(ENDPOINT_NUM, &self.out_buffer[0].buf);
+         self.controller()
+-            .endpoint_in_out_enable(TransferType::Interrupt, 1);
++            .endpoint_in_out_enable(TransferType::Interrupt, ENDPOINT_NUM);
++        // Set up the vendor in-out endpoint
++        #[cfg(feature = "vendor_hid")]
++        {
++            self.controller()
++                .endpoint_set_in_buffer(VENDOR_ENDPOINT_NUM, &self.in_buffer[1].buf);
++            self.controller()
++                .endpoint_set_out_buffer(VENDOR_ENDPOINT_NUM, &self.out_buffer[1].buf);
++            self.controller()
++                .endpoint_in_out_enable(TransferType::Interrupt, VENDOR_ENDPOINT_NUM);
++        }
```

Comment:
Shouldn't this be a loop from 1 to NUM_ENDPOINTS?

Reply:
Indeed. Fixed.